api/v1alpha1/observatorium_types.go

```diff
@@ -0,0 +1,143 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ObservatoriumSpec defines the desired state of Observatorium
+type ObservatoriumSpec struct {
+	// Thanos Image name
+	Image string `json:"thanosImage"`
+	// Thanos Image version
+	Version string `json:"thanosVersion"`
+	// Objest Storage Configuration
+	ObjectStorageConfig ObjectStorageConfig `json:"objectStorageConfig"`
+	// Hashrings describes a list of Hashrings
+	Hashrings []*Hashring `json:"hashrings"`
+	// Thanos CompactSpec
+	Compact CompactSpec `json:"compact"`
+	// Thanos Receive Controller Spec
+	ReceiveController ReceiveControllerSpec `json:"thanosReceiveController"`
+	// Thanos ThanosPersistentSpec
+	Receivers ReceiversSpec `json:"receivers"`
+	// Thanos QuerierCache
+	QueryCache QuerierCacheSpec `json:"queryCache"`
```

Comment:
We should try to be consistent here with naming.

Above, we use `CompactSpec` and `ReceiveSpec` but here we use `QuerierCacheSpec`. Shouldn't it be `QueryCacheSpec`?

Reply:
Done